src/reducers/fileTree.tsx

```diff
@@ -191,6 +190,100 @@ export const getRelativePath = ({
   return pathList[newIndex];
 };
 
+const findRelativeMessageUid = (
+  currentPath: string,
+  messageMap: LinterMessageMap,
+  pathList: string[],
+  position: RelativePathPosition,
+): LinterMessage['uid'] | null => {
+  let path = currentPath;
+  for (let i = 0; i < pathList.length; i++) {
+    const nextPath = getRelativePath({
+      currentPath: path,
+      pathList,
+      position,
+    });
+    const nextMessages = messageMap[nextPath];
+    if (nextMessages) {
+      const nextMessagesForPath = getMessagesForPath(nextMessages);
+      if (position === RelativePathPosition.previous) {
+        const lastMessageIndex = nextMessagesForPath.length - 1;
+        if (lastMessageIndex >= 0) {
+          return nextMessagesForPath[lastMessageIndex].uid;
+        }
+      } else if (nextMessagesForPath.length) {
+        return nextMessagesForPath[0].uid;
+      }
+    }
+    path = nextPath;
+  }
+  // If we got here then we never found another message, which would be odd,
+  // because we should have at the very least gotten back to our original
+  // message.
+  // We should probably log this.
+  return null;
```

Comment:
It probably makes more sense to throw an error here, as this should never happen. I'm not sure how I would be able to test this though, again because it shouldn't be possible.

Reply:
> It probably makes more sense to throw an error here, as this should never happen. I'm not sure how I would be able to test this though, again because it shouldn't be possible.

You should throw an error, if possible.

A test using an empty `pathList` will make it throw:

```diff
diff --git a/src/reducers/fileTree.spec.tsx b/src/reducers/fileTree.spec.tsx
index 925c39e..6061b2b 100644
--- a/src/reducers/fileTree.spec.tsx
+++ b/src/reducers/fileTree.spec.tsx
@@ -932,6 +932,27 @@ describe(__filename, () => {
         ).toEqual(global11);
       });
 
+      it('throws...', () => {
+        const externalMessages1 = [
+          { line: null, uid: global11 },
+          { line: null, uid: global21 },
+        ];
+        const externalMessages2 = [{ line: null, uid: global12 }];
+
+        const messageMap = createMessageMap([
+          { path: file1, messages: externalMessages1 },
+          { path: file3, messages: externalMessages2 },
+        ]);
+
+        _getRelativeMessageUid({
+          currentMessageUid: global12,
+          currentPath: file3,
+          messageMap,
+          pathList: [],
+          position: RelativePathPosition.next,
+        });
+      });
+
       it('wraps around from the first message in the first file to the last message in the last file', () => {
         const externalMessages1 = [
           { line: null, uid: global11 },
```